h2o-py/h2o/explanation/_explain.py

```diff
@@ -1362,6 +1396,11 @@ def ice_plot(
             plt.savefig(fname=save_plot_path)
         return decorate_plot_result(figure=fig)
 
+def _isnan(value):
+    if isinstance(value, float):
+        return np.isnan(value)
+    else:
+        return False
```

Comment:
I consider this to be a little fishy because basically anything that is not a _float_ is automatically a _NaN_. It seems that more robust would be to at least try casting the value to float, and only then perform the check.

Reply:
only `orig_value` is passed into this method (`orig_value = frame.as_data_frame()[column][index]`).

`as_data_frame() `reads the dataset as a python local object and it behaves like: 

- if there is a missing value present in certain numerical column, then the resulting  `frame.as_data_frame()` contains this column with type 'float'. Even if it is of type int in `frame`. Because python NaN is defined as a special floating-point value and cannot be converted to any other type than float.

- if there is or is not a missing value present in certain factor column (binary or other enumeration), then the resulting `frame.as_data_frame()` contains this column with type 'object'. On indexes with non-missing values it contains strings with categories, on indexes with missing values it contains NaN (of type float), hence, if the orig_value (= frame.as_data_frame()[column][index]) is not a float, I can safely say it is not a NaN. Being float is a prerequisite to being NaN here. But trying to cast a string value to float, for example float("Class 1"), will certainly result in a ValueError.